### ISSUE TITLE
[@azure/msal-browser@2.0.0-beta.2] B2C & knownAuthorities not working

### ISSUE THREAD

**Issue by known-as-bmf** [AzureAD/microsoft-authentication-library-for-js#1683]
## Library
- [ ] `msal@1.x.x` or `@azure/msal@1.x.x`
- [x] `@azure/msal-browser@2.x.x`
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

**Important**: Please fill in your exact version number above

`@azure/msal-browser@2.0.0-beta.2`

## Framework
`react@16.13.1` (irrelevant)

## Description

After creating a new `PublicClientApplication` with a custom b2c authority and this authority in `knownAuthorities`, the client fails during logins, trying to validate the authority (calling https://login.microsoftonline.com/common/discovery/instance?api-version=1.0&authorization_endpoint=[authority] even tho the autority is in `knownAuthorities` in the conf).

## Configuration

```ts
const msalConfig: Configuration = {
  auth: {
    clientId: '[redacted]',
    authority: 'https://[tenant name].b2clogin.com/[tenant name].onmicrosoft.com/[policy]',
    knownAuthorities: ['[tenant name].b2clogin.com'],
  },
  cache: { cacheLocation: 'localStorage', storeAuthStateInCookie: false },
};
```

## Reproduction steps

First, the call stack where the problem occurs:

> detectAuthorityFromUrl (d:/Dev/msal-common/dist/index.es.js#1835)
createInstance (d:/Dev/msal-common/dist/index.es.js#1854)
BaseClient (d:/Dev/msal-common/dist/index.es.js#1887)
SPAClient (d:/Dev/msal-common/dist/index.es.js#2947)
constructor (d:/Dev/ts/src/app/PublicClientApplication.ts#78)

Explanation:
1. `PublicClientApplication` creates a `SPAClient`
![image](https://user-images.githubusercontent.com/11721166/82455931-ddd88700-9ab3-11ea-8d59-3485a2bbaeed.png)
2. `SPAClient` invokes its base ctor (ignore the breakpoint)
![image](https://user-images.githubusercontent.com/11721166/82456046-fd6faf80-9ab3-11ea-8d77-204b6d0253fc.png)
3. The base ctor creates an `Authority`
![image](https://user-images.githubusercontent.com/11721166/82456115-16786080-9ab4-11ea-971a-eb01f7c1cd08.png)
4. While being created, the authority tries to detect its type
![image](https://user-images.githubusercontent.com/11721166/82456253-49225900-9ab4-11ea-8b45-fd119093cc4e.png)
5. This detection looks for b2c known hosts
![image](https://user-images.githubusercontent.com/11721166/82456318-62c3a080-9ab4-11ea-8a40-5b4d57ef351c.png)
6. Doesnt find any and resolves to `AuthorityType.Aad`.

The problem is that `B2CAuthority.B2CTrustedHostList` is empty at this point.

`B2cAuthority.setKnownAuthorities(_this.config.authOptions.knownAuthorities);` in `SPAClient` is only called after the authority has already been created (look at step 2).

The Authority will always be of kind `Aad` (except for the `Adfs` special case). This authority kind checks the validity of the authority agains microsoft discovery service, which fails.

## Expected behavior
`knownAuthorities` taken into account on `PublicClientApplication` creation.

## Browsers
Not browser specific, but Firefox 76.0.1 (64 bits) used in my tests

**Comment by known-as-bmf** [comment]
From the codebase, looks like it's fixed but not released.

Is there any plan of releasing a new version soon ?

**Comment by pkanher617** [comment]
@known-as-bmf I will push out a new release as soon as we have merged in #1675.

**Comment by tnorling** [comment]
@known-as-bmf `msal-browser@2.0.0-beta.3` has been released and should fix this issue. Please be aware there is currently an issue with the b2c service which is being tracked in #1795 . Closing this as the original issue has been resolved.

**Comment by vaughanh1** [comment]
I have mentioned this issue due to receiving this error with the latest version 2.0.0 of msal 2 (msal-browser). Please see my initial comment in:
https://github.com/MicrosoftDocs/azure-docs/issues/59656

**Comment by tnorling** [comment]
@vaughanh1 Please open a new issue if you are not able to get this to work. As mentioned above this was fixed in version 2.0.0-beta.3

**Comment by vaughanh1** [comment]
Not to worry. Thanks to @tnorling, he pointed we out that my configuration of the know authorities field should not have the 'https://' at the start. Thanks again.
